unsigned int tomo_db0::IoLoop::epoll_pass<true, false, true>(int)
--- PRE

+++ POST

@@ -2,7 +2,7 @@

 0004 [ 2] push %r15
 0006 [ 2] push %r14
 0008 [ 2] push %r13
-000a [ 3] mov %rdi,%r15
+000a [ 3] mov %rdi,%r14
 000d [ 2] push %r12
 000f [ 1] push %rbp
 0010 [ 1] push %rbx
@@ -20,182 +20,177 @@

 0044 [ 8] movl $0x0,0xc(%rsp)
 004c [ 2] test %eax,%eax
 004e [ 4] cmovns %eax,%r12d
-0052 [ 7] mov %r12d,0xc18(%r15)
-0059 [ 6] jle <+0x11f>
-005f [ 7] mov 0x190(%r15),%rax
-0066 [ 3] movslq %r12d,%rdx
-0069 [ 2] xor %ebp,%ebp
-006b [ 7] lea <rip>(%rip),%r13 # <typeinfo name for std::_Sp_counted_ptr_inplace<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>+0xadc>
-0072 [ 7] add %rdx,0x520(%rax)
-0079 [ 7] lea 0xac8(%r15),%rax
-0080 [ 5] mov %rax,0x10(%rsp)
-0085 [11] data16 cs nopw 0x0(%rax,%rax,1)
-0090 [ 4] mov 0x4(%rbx),%rdx
-0094 [ 3] mov %rdx,%rax
+0052 [ 7] mov %r12d,0xc18(%r14)
+0059 [ 2] jle <+0xcb>
+005b [ 7] mov 0x190(%r14),%rax
+0062 [ 3] movslq %r12d,%rdx
+0065 [ 2] xor %ebp,%ebp
+0067 [ 7] lea <rip>(%rip),%r13 # <typeinfo name for std::_Sp_counted_ptr_inplace<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>+0xadc>
+006e [ 7] add %rdx,0x520(%rax)
+0075 [ 7] lea 0xac8(%r14),%rax
+007c [ 5] mov %rax,0x10(%rsp)
+0081 [11] data16 cs nopw 0x0(%rax,%rax,1)
+008c [ 4] nopl 0x0(%rax)
+0090 [ 4] mov 0x4(%rbx),%rsi
+0094 [ 3] mov %rsi,%rax
 0097 [ 4] shr $0x38,%rax
 009b [ 2] cmp $0x11,%al
-009d [ 2] ja <+0x110>
+009d [ 2] ja <+0xb0>
 009f [ 5] movslq 0x0(%r13,%rax,4),%rax
 00a4 [ 3] add %r13,%rax
 00a7 [ 3] notrack jmp *%rax
-00aa [ 2] mov (%rbx),%eax
-00ac [ 5] test $0x2019,%eax
-00b1 [ 2] je <+0xb7>
-00b3 [ 4] movb $0x0,0x29(%rdx)
-00b7 [ 2] test $0x1c,%al
-00b9 [ 2] je <+0xe8>
-00bb [ 4] cmpb $0x0,0x30(%rdx)
-00bf [ 5] mov %rdx,0x20(%rsp)
-00c4 [ 2] jne <+0xe8>
-00c6 [ 5] lea 0x20(%rsp),%r14
-00cb [ 7] lea 0x198(%r15),%rdi
-00d2 [ 4] movb $0x1,0x30(%rdx)
-00d6 [ 5] mov %rdx,0x18(%rsp)
-00db [ 3] mov %r14,%rsi
-00de [ 5] call <std::deque<tomo_db0::Client*, std::allocator<tomo_db0::Client*> >::push_back(tomo_db0::Client* const&)>
-00e3 [ 5] mov 0x18(%rsp),%rdx
-00e8 [ 3] mov %rdx,%rsi
-00eb [ 3] mov %r15,%rdi
-00ee [ 5] call <tomo_db0::IoLoop::mark_active(tomo_db0::Client*)>
-00f3 [11] data16 cs nopw 0x0(%rax,%rax,1)
-00fe [ 2] xchg %ax,%ax
-0100 [ 4] incl 0xc(%rsp)
-0104 [11] data16 cs nopw 0x0(%rax,%rax,1)
-010f [ 1] nop
-0110 [ 2] inc %ebp
-0112 [ 4] add $0xc,%rbx
-0116 [ 3] cmp %r12d,%ebp
-0119 [ 6] jl <+0x90>
-011f [ 5] mov 0x28(%rsp),%rax
-0124 [ 9] sub %fs:0x28,%rax
-012d [ 6] jne <+0x368>
-0133 [ 4] mov 0xc(%rsp),%eax
-0137 [ 4] add $0x38,%rsp
-013b [ 1] pop %rbx
-013c [ 1] pop %rbp
-013d [ 2] pop %r12
-013f [ 2] pop %r13
-0141 [ 2] pop %r14
-0143 [ 2] pop %r15
-0145 [ 1] ret
-0146 [10] cs nopw 0x0(%rax,%rax,1)
-0150 [ 7] mov 0x190(%r15),%rax
-0157 [ 5] mov 0x10(%rsp),%rdx
-015c [ 7] lea <rip>(%rip),%rdi # <tomo_db0::Server::databases_>
-0163 [ 2] inc %ebp
-0165 [ 4] add $0xc,%rbx
-0169 [ 2] mov (%rax),%esi
-016b [ 5] call <tomo_db0::DatabaseMap::consume_wake(unsigned int, tomo_db0::Ring&)>
-0170 [ 4] incl 0xc(%rsp)
-0174 [ 3] cmp %r12d,%ebp
-0177 [ 6] jl <+0x90>
-017d [ 2] jmp <+0x11f>
-017f [ 1] nop
-0180 [ 7] mov 0x188(%r15),%rax
-0187 [ 7] movzbl 0x11460(%rax),%eax
-018e [ 2] test %al,%al
-0190 [ 6] jne <+0x110>
-0196 [ 7] mov 0xaa8(%r15),%r14d
-019d [ 3] test %r14d,%r14d
-01a0 [ 6] js <+0x110>
-01a6 [ 8] movl $0x0,0x18(%rsp)
-01ae [ 2] xor %edx,%edx
-01b0 [ 2] xor %esi,%esi
-01b2 [ 5] mov $0x80800,%ecx
-01b7 [ 3] mov %r14d,%edi
-01ba [ 5] call <accept4@plt>
-01bf [ 2] test %eax,%eax
-01c1 [ 6] jns <+0x350>
-01c7 [ 5] call <__errno_location@plt>
-01cc [ 2] mov (%rax),%eax
-01ce [ 3] cmp $0x4,%eax
-01d1 [ 6] jne <+0x2ff>
-01d7 [ 2] jmp <+0x1ae>
-01d9 [ 7] nopl 0x0(%rax)
-01e0 [ 5] lea 0x20(%rsp),%r14
-01e5 [ 9] movq $0x0,0x20(%rsp)
+00aa [ 6] nopw 0x0(%rax,%rax,1)
+00b0 [ 3] mov %r14,%rdi
+00b3 [ 5] call <tomo_db0::IoLoop::fliplisten_epoll(unsigned long)>
+00b8 [ 4] add %eax,0xc(%rsp)
+00bc [ 4] nopl 0x0(%rax)
+00c0 [ 2] inc %ebp
+00c2 [ 4] add $0xc,%rbx
+00c6 [ 3] cmp %r12d,%ebp
+00c9 [ 2] jl <+0x90>
+00cb [ 5] mov 0x28(%rsp),%rax
+00d0 [ 9] sub %fs:0x28,%rax
+00d9 [ 6] jne <+0x348>
+00df [ 4] mov 0xc(%rsp),%eax
+00e3 [ 4] add $0x38,%rsp
+00e7 [ 1] pop %rbx
+00e8 [ 1] pop %rbp
+00e9 [ 2] pop %r12
+00eb [ 2] pop %r13
+00ed [ 2] pop %r14
+00ef [ 2] pop %r15
+00f1 [ 1] ret
+00f2 [11] data16 cs nopw 0x0(%rax,%rax,1)
+00fd [ 3] nopl (%rax)
+0100 [10] movabs $0xffffffffffff,%rax
+010a [ 3] and %rsi,%rax
+010d [ 2] je <+0xc0>
+010f [ 6] mov 0x788(%rax),%edx
+0115 [ 7] mov 0x190(%r14),%rcx
+011c [ 2] cmp %edx,(%rcx)
+011e [ 2] jne <+0xc0>
+0120 [ 4] cmpb $0x0,0x33(%rax)
+0124 [ 2] jne <+0xc0>
+0126 [ 6] mov 0x784(%rax),%edx
+012c [ 3] cmp $0xfffffffe,%edx
+012f [ 2] je <+0xc0>
+0131 [ 7] mov 0x288(%r14),%rdx
+0138 [ 7] mov 0x290(%r14),%rcx
+013f [ 3] cmp %rcx,%rdx
+0142 [ 2] je <+0x162>
+0144 [11] data16 cs nopw 0x0(%rax,%rax,1)
+014f [ 1] nop
+0150 [ 3] cmp (%rdx),%rax
+0153 [ 6] je <+0xc0>
+0159 [ 4] add $0x28,%rdx
+015d [ 3] cmp %rdx,%rcx
+0160 [ 2] jne <+0x150>
+0162 [ 2] mov (%rbx),%edx
+0164 [ 6] test $0x2019,%edx
+016a [ 2] je <+0x170>
+016c [ 4] movb $0x0,0x29(%rax)
+0170 [ 3] and $0x1c,%edx
+0173 [ 2] je <+0x1a2>
+0175 [ 4] cmpb $0x0,0x30(%rax)
+0179 [ 5] mov %rax,0x20(%rsp)
+017e [ 2] jne <+0x1a2>
+0180 [ 5] lea 0x20(%rsp),%r15
+0185 [ 7] lea 0x198(%r14),%rdi
+018c [ 4] movb $0x1,0x30(%rax)
+0190 [ 5] mov %rax,0x18(%rsp)
+0195 [ 3] mov %r15,%rsi
+0198 [ 5] call <std::deque<tomo_db0::Client*, std::allocator<tomo_db0::Client*> >::push_back(tomo_db0::Client* const&)>
+019d [ 5] mov 0x18(%rsp),%rax
+01a2 [ 3] mov %rax,%rsi
+01a5 [ 3] mov %r14,%rdi
+01a8 [ 5] call <tomo_db0::IoLoop::mark_active(tomo_db0::Client*)>
+01ad [ 3] nopl (%rax)
+01b0 [ 4] incl 0xc(%rsp)
+01b4 [ 5] jmp <+0xc0>
+01b9 [ 7] nopl 0x0(%rax)
+01c0 [ 7] mov 0x190(%r14),%rax
+01c7 [ 5] mov 0x10(%rsp),%rdx
+01cc [ 7] lea <rip>(%rip),%rdi # <tomo_db0::Server::databases_>
+01d3 [ 2] mov (%rax),%esi
+01d5 [ 5] call <tomo_db0::DatabaseMap::consume_wake(unsigned int, tomo_db0::Ring&)>
+01da [ 4] incl 0xc(%rsp)
+01de [ 5] jmp <+0xc0>
+01e3 [11] data16 cs nopw 0x0(%rax,%rax,1)
 01ee [ 2] xchg %ax,%ax
-01f0 [ 7] mov 0xbc8(%r15),%edi
-01f7 [ 5] mov $0x8,%edx
-01fc [ 3] mov %r14,%rsi
-01ff [ 5] call <read@plt>
-0204 [ 4] cmp $0x8,%rax
-0208 [ 2] je <+0x1f0>
-020a [ 7] mov 0x190(%r15),%rax
-0211 [ 2] inc %ebp
-0213 [ 4] add $0xc,%rbx
-0217 [ 4] incl 0xc(%rsp)
-021b [ 7] incq 0x440(%rax)
-0222 [ 3] cmp %r12d,%ebp
-0225 [ 6] jl <+0x90>
-022b [ 5] jmp <+0x11f>
-0230 [10] movabs $0xffffffffffff,%rax
-023a [ 3] and %rax,%rdx
-023d [ 6] je <+0x110>
-0243 [ 6] mov 0x788(%rdx),%eax
-0249 [ 7] mov 0x190(%r15),%rcx
-0250 [ 2] cmp %eax,(%rcx)
-0252 [ 6] jne <+0x110>
-0258 [ 4] cmpb $0x0,0x33(%rdx)
-025c [ 6] jne <+0x110>
-0262 [ 6] mov 0x784(%rdx),%eax
-0268 [ 3] cmp $0xfffffffe,%eax
-026b [ 6] je <+0x110>
-0271 [ 7] mov 0x288(%r15),%rax
-0278 [ 7] mov 0x290(%r15),%rcx
-027f [ 3] cmp %rcx,%rax
-0282 [ 2] jne <+0x29d>
-0284 [ 5] jmp <+0xaa>
-0289 [ 7] nopl 0x0(%rax)
-0290 [ 4] add $0x28,%rax
-0294 [ 3] cmp %rax,%rcx
-0297 [ 6] je <+0xaa>
-029d [ 3] cmp (%rax),%rdx
-02a0 [ 2] jne <+0x290>
-02a2 [ 5] jmp <+0x110>
-02a7 [ 9] nopw 0x0(%rax,%rax,1)
-02b0 [ 7] mov 0x188(%r15),%rax
-02b7 [ 7] movzbl 0x11460(%rax),%eax
-02be [ 2] test %al,%al
-02c0 [ 6] jne <+0x110>
-02c6 [ 7] mov 0xaa0(%r15),%r14d
-02cd [ 3] test %r14d,%r14d
-02d0 [ 6] js <+0x110>
-02d6 [ 8] movl $0x0,0x18(%rsp)
-02de [ 2] xor %edx,%edx
-02e0 [ 2] xor %esi,%esi
-02e2 [ 5] mov $0x80800,%ecx
-02e7 [ 3] mov %r14d,%edi
-02ea [ 5] call <accept4@plt>
-02ef [ 2] test %eax,%eax
-02f1 [ 2] jns <+0x330>
-02f3 [ 5] call <__errno_location@plt>
-02f8 [ 2] mov (%rax),%eax
-02fa [ 3] cmp $0x4,%eax
-02fd [ 2] je <+0x2de>
-02ff [ 3] cmp $0xb,%eax
-0302 [ 2] je <+0x312>
-0304 [ 7] mov 0x190(%r15),%rax
-030b [ 7] incq 0x458(%rax)
-0312 [ 4] mov 0x18(%rsp),%ecx
-0316 [ 2] inc %ebp
-0318 [ 4] add $0xc,%rbx
-031c [ 4] add %ecx,0xc(%rsp)
-0320 [ 3] cmp %r12d,%ebp
-0323 [ 6] jl <+0x90>
-0329 [ 5] jmp <+0x11f>
-032e [ 2] xchg %ax,%ax
-0330 [ 5] mov $0x1,%edx
+01f0 [ 7] mov 0x188(%r14),%rax
+01f7 [ 7] movzbl 0x11460(%rax),%eax
+01fe [ 2] test %al,%al
+0200 [ 6] jne <+0xc0>
+0206 [ 7] mov 0xaa8(%r14),%r15d
+020d [ 3] test %r15d,%r15d
+0210 [ 6] js <+0xc0>
+0216 [ 8] movl $0x0,0x18(%rsp)
+021e [ 2] xor %edx,%edx
+0220 [ 2] xor %esi,%esi
+0222 [ 5] mov $0x80800,%ecx
+0227 [ 3] mov %r15d,%edi
+022a [ 5] call <accept4@plt>
+022f [ 2] test %eax,%eax
+0231 [ 6] jns <+0x330>
+0237 [ 5] call <__errno_location@plt>
+023c [ 2] mov (%rax),%eax
+023e [ 3] cmp $0x4,%eax
+0241 [ 6] jne <+0x2ef>
+0247 [ 2] jmp <+0x21e>
+0249 [ 7] nopl 0x0(%rax)
+0250 [ 5] lea 0x20(%rsp),%r15
+0255 [ 9] movq $0x0,0x20(%rsp)
+025e [ 2] xchg %ax,%ax
+0260 [ 7] mov 0xbc8(%r14),%edi
+0267 [ 5] mov $0x8,%edx
+026c [ 3] mov %r15,%rsi
+026f [ 5] call <read@plt>
+0274 [ 4] cmp $0x8,%rax
+0278 [ 2] je <+0x260>
+027a [ 7] mov 0x190(%r14),%rax
+0281 [ 4] incl 0xc(%rsp)
+0285 [ 7] incq 0x440(%rax)
+028c [ 5] jmp <+0xc0>
+0291 [11] data16 cs nopw 0x0(%rax,%rax,1)
+029c [ 4] nopl 0x0(%rax)
+02a0 [ 7] mov 0x188(%r14),%rax
+02a7 [ 7] movzbl 0x11460(%rax),%eax
+02ae [ 2] test %al,%al
+02b0 [ 6] jne <+0xc0>
+02b6 [ 7] mov 0xaa0(%r14),%r15d
+02bd [ 3] test %r15d,%r15d
+02c0 [ 6] js <+0xc0>
+02c6 [ 8] movl $0x0,0x18(%rsp)
+02ce [ 2] xor %edx,%edx
+02d0 [ 2] xor %esi,%esi
+02d2 [ 5] mov $0x80800,%ecx
+02d7 [ 3] mov %r15d,%edi
+02da [ 5] call <accept4@plt>
+02df [ 2] test %eax,%eax
+02e1 [ 2] jns <+0x310>
+02e3 [ 5] call <__errno_location@plt>
+02e8 [ 2] mov (%rax),%eax
+02ea [ 3] cmp $0x4,%eax
+02ed [ 2] je <+0x2ce>
+02ef [ 3] cmp $0xb,%eax
+02f2 [ 2] je <+0x302>
+02f4 [ 7] mov 0x190(%r14),%rax
+02fb [ 7] incq 0x458(%rax)
+0302 [ 4] mov 0x18(%rsp),%ecx
+0306 [ 4] add %ecx,0xc(%rsp)
+030a [ 5] jmp <+0xc0>
+030f [ 1] nop
+0310 [ 5] mov $0x1,%edx
+0315 [ 2] mov %eax,%esi
+0317 [ 3] mov %r14,%rdi
+031a [ 4] incl 0x18(%rsp)
+031e [ 5] call <void tomo_db0::IoLoop::admit_fd<true, true>(int, tomo_db0::UrKind)>
+0323 [ 2] jmp <+0x2ce>
+0325 [11] data16 cs nopw 0x0(%rax,%rax,1)
+0330 [ 5] mov $0x6,%edx
 0335 [ 2] mov %eax,%esi
-0337 [ 3] mov %r15,%rdi
+0337 [ 3] mov %r14,%rdi
 033a [ 4] incl 0x18(%rsp)
 033e [ 5] call <void tomo_db0::IoLoop::admit_fd<true, true>(int, tomo_db0::UrKind)>
-0343 [ 2] jmp <+0x2de>
-0345 [11] data16 cs nopw 0x0(%rax,%rax,1)
-0350 [ 5] mov $0x6,%edx
-0355 [ 2] mov %eax,%esi
-0357 [ 3] mov %r15,%rdi
-035a [ 4] incl 0x18(%rsp)
-035e [ 5] call <void tomo_db0::IoLoop::admit_fd<true, true>(int, tomo_db0::UrKind)>
-0363 [ 5] jmp <+0x1ae>
-0368 [ 5] call <__stack_chk_fail@plt>
+0343 [ 5] jmp <+0x21e>
+0348 [ 5] call <__stack_chk_fail@plt>
